aodndata/moorings/products_handler.py

```diff
@@ -75,19 +95,12 @@ def _get_input_files(self):
                        PropertyIsNotEqualTo(propertyname='data_category', literal='CTD_profiles'),
                        PropertyIsNotEqualTo(propertyname='data_category', literal='aggregated_timeseries')
                        ]
-        ogc_filter = ogc_filter_to_string(And(filter_list))
-        # Note I need to access _wfs_broker to be able to use query_urls_for_layer() with a filter,
-        # as the corresponding StateQuery method doesn't accept additional kwargs.
-        # TODO: find out why this calls getCapabilities twice (and takes 40s even when response mocked with httpretty)
-        # TODO: replace ._wfs_broker.getfeature_dict() with .getfeature_dict() once aodncore has been updated
-        wfs_response = self.state_query._wfs_broker.getfeature_dict(typename=[self.FILE_INDEX_LAYER],
-                                                                    filter=ogc_filter,
-                                                                    propertyname=['url', 'variables']
-                                                                    )
+        wfs_features = self._get_wfs_features(filter_list, propertyname=['url', 'variables'])
         self.input_file_variables = {f['properties']['url']: f['properties']['variables'].split(', ')
```

Comment:
Same comment as the one before - just check for emptyness of wfs_features explicitly.

Reply:
I mean why you should set a empty dictionary with a dictionary comprehensions with a for loop? Just check if was features is empty and raise the exception...

Reply:
Ok. Check added.